### DISCUSSION TITLE
Reference frame clarification

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#102] [Q&A] [answered]
Hello,
 I have few questions with regards to reference frames in GROOPS and more specifically for PPP processing. In which reference frame is the station after the GNSSProcessing script is done? In the plot example there is a part of the program where 2 rotations are done. For LocalLevelFrame2StarCamera it states that the rotation is done from ellipsoidal north, east, up (output of processing)  to TRF - terrestrial reference frame. Which reference frame is the TRF then exactly? ITRF14,.... Which additional rotation does the InstrumentRotate program do afterwards (the documentation states ... satellite frame to inertial frame)? 

The errorbars on the plots are based on covariance values - is there any transformation needed as well for those values? The LocalLevelFrame2StarCamera only uses the first 3 columns of the output position file as per the documentation. So if there is a transformation needed how would this be achieved?

We want to compare our station position results to the University of Nevada station results (for example: station bas4) and their results are shown in IGS14 format - is there an possibility in GROOPS to get results directly in IGS 14 for example or in any other standard reference frame? 

Thank you for your help and for your continued help

Cheers
Jean

PS - Thank you for posting the videos - they are very helpful to get started on GROOPS

**Comment by tmayerguerr** [comment, marked as answer]
Hi Jean,

The positions are estimated in a global terrestrial reference frame, which is defined by the orbits. In the example script the TUG Repro3 orbits are used, therefore the reference frame is that of igsR3, which itself is aligned to IGb14.

The program LocalLevelFrame2StarCamera computes the rotation matrix from local system (North, east, down) to a global coordinate system at a given ellipsoidal point. The second program InstrumentRotate applies this rotation matrices to the positions. The file format was originally intended to store star camera observations (attitude) of LEO satellites. Hence the name of the program.

The computed output/{loopTime:%D}/covarianceEpoch3x3_{loopTime:%D}.{station}.dat are given in the same global reference system as the positions. To get the variances in a local system you should apply InstrumentRotate with same rotation file ({tmpDir}/local2trf_{loopTime:%D}.{station}.dat) to the covarianceEpoch3x3 (input/outputfileInstrument).

There is no program in GROOPS to apply reference frame transformations directly at the moment. We will think about it.

regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

TXS for the quick reply

Cheers,
Jean
